rotkehlchen/types.py

```diff
@@ -658,23 +660,90 @@ def serialize(self) -> dict[str, Optional[str]]:
             'blockchain': self.blockchain.serialize() if self.blockchain is not None else None,
         }
 
-    def serialize_for_db(self) -> tuple[str, str, Optional[str]]:
+    def serialize_for_db(self) -> tuple:
         blockchain = self.blockchain.value if self.blockchain is not None else None
         return (self.address, self.name, blockchain)
 
     @classmethod
-    def deserialize(cls: type['AddressbookEntry'], data: dict[str, Any]) -> 'AddressbookEntry':
+    def deserialize_from_db(cls, entry: tuple) -> 'NamedAddressbookEntry':
+        return cls(
+            address=entry[0],
+            name=entry[1],
+            blockchain=SupportedBlockchain(entry[2]) if entry[2] is not None else None,
+        )
+
+    @classmethod
+    def deserialize(cls: type['NamedAddressbookEntry'], data: dict[str, Any]) -> 'NamedAddressbookEntry':  # noqa: E501
+        """May raise:
+        - KeyError if required keys are missing
+        - DeserializationError if blockchain is not known
+        """
+        return cls(
+            address=data['address'],
+            name=data['name'],
+            blockchain=SupportedBlockchain.deserialize(data['blockchain']) if data['blockchain'] is not None else None,  # noqa: E501
+        )
+
+    def __str__(self) -> str:
+        return f'Named addressbook entry with name "{self.name}", address "{self.address}" and blockchain {self.blockchain.value if self.blockchain is not None else None}'  # noqa: E501
```

Comment:
since is the same string maybe you could have this in a method called `_base_str` and then in the subclass call 

```
return super()._base_str.format(book_name='your book name')
```